tests/PHPStan/Analyser/data/preg_match_shapes.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace PregMatchShapes;
+
+use function PHPStan\Testing\assertType;
+
```

Comment:
When the return value of preg_match is false, the type of $matches is always null:

```suggestion
function invalid(string $s): void
{
    preg_match('', $s, $matches);

    assertType('null', $matches);
}
```

Reply:
these `null` expecting cases are interessting, but I feel out of scope of this PR.

the case of explicitly handling a possible syntax error in the pattern in application code usually does not happen.
even before this PR phpstan atm does not make `$matches` return null

Reply:
please open a new separate phpstan issue for these cases with a snippet and expectations